lib/ffi-compiler/compile_task.rb

```diff
@@ -143,14 +148,25 @@ def define_task!
           index += 1
         end
 
-        ld = src_files.detect { |f| f =~ /\.cpp$/ } ? cxx : cc
+        ld = if src_files.detect { |f| f =~ /\.cpp$/ }
+               cxx
+             elsif src_files.detect { |f| f =~ /\.c$/ }
+               cc
+             else
+               go
+             end
 
         # create all the directories for the output files
         obj_files.map { |f| File.dirname(f) }.sort.uniq.map { |d| directory d }
 
         desc "Build dynamic library"
         file lib_name => obj_files do |t|
-          sh "#{ld} #{so_flags} -o #{t.name} #{t.prerequisites.join(' ')} #{ld_flags} #{libs}"
+          if ld == cxx || ld == cc
+            sh "#{ld} #{so_flags} -o #{t.name} #{t.prerequisites.join(' ')} #{ld_flags} #{libs}"
+          else
+            # this is a go library, which is already a shared C lib; copy it into place
+            sh "cp #{t.prerequisites[0]} #{t.name}"
```

Comment:
why not use `FileUtils.cp` ? I think that would be better than shelling out just to copy a file.